lib/IRGen/GenDecl.cpp

```diff
@@ -1132,6 +1133,13 @@ void IRGenerator::emitLazyDefinitions() {
       CurrentIGMPtr IGM = getGenModule(type->getDeclContext());
       emitLazyTypeMetadata(*IGM.get(), type);
     }
+    while (!LazySpecializedTypeMetadataRecords.empty()) {
+      CanBoundGenericType type =
```

Comment:
What about pre-specializing types such as `Foo<T>.Bar`?

Reply:
Should work now.